Classes/Service/ImportService.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SUDHAUS7\Xlsimport\Service;
+
+use Doctrine\DBAL\DBALException;
+use Psr\EventDispatcher\EventDispatcherInterface;
+use SUDHAUS7\Xlsimport\Domain\Dto\ImportJob;
+use SUDHAUS7\Xlsimport\Event\ManipulateRelationsEvent;
+use TYPO3\CMS\Core\Core\Environment;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\DataHandling\DataHandler;
+use TYPO3\CMS\Core\Resource\Exception\FileDoesNotExistException;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Utility\StringUtility;
+
+/**
+ * @internal This class is for internal usage and no Public API
+ * Code can change in future versions
+ */
+final class ImportService
+{
+    protected EventDispatcherInterface $eventDispatcher;
+
+    public function __construct(
+        EventDispatcherInterface $eventDispatcher
+    ) {
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function isImportAllowed(string $table): bool
+    {
+        return array_key_exists($table, $GLOBALS['TCA']);
+    }
+
+    /**
+     * @throws \JsonException
+     * @throws FileDoesNotExistException
+     */
+    public function prepareImport(ImportJob $importJob): void
+    {
+        $insertData = [
+            $importJob->getTable() => [],
+        ];
+
+        $importData = $this->loadDataFromJsonFile($importJob->getJsonFile());
+
+        foreach ($importData as $importPosition => $data) {
+            // cleanup array from not importable data lines
+            if (
+                !array_key_exists($importPosition, $importJob->getDatasetMapping())
+                || $importJob->getDatasetMapping()[$importPosition] !== '1'
+            ) {
+                unset($importData[$importPosition]);
+                continue;
+            }
+            $insertArray = [];
+            $update = false;
+            foreach ($importJob->getFieldMapping() as $position => $field) {
+                if ($field === 'uid') {
+                    if (!empty($data[$position])) {
+                        $update = true;
+                    }
+                } else {
+                    $insertArray[$field] = $data[$position];
+                }
+            }
+            if (!isset($insertArray['pid'])) {
+                $insertArray['pid'] = $importJob->getPid();
+            }
+            foreach ($importJob->getDefaultValues() as $fieldName => $defaultValue) {
+                $insertArray[$fieldName] = $defaultValue;
+            }
+
+            foreach ($importJob->getPasswordFields() as $passwordField) {
+                $insertArray[$passwordField] = md5(sha1(microtime()));
+            }
+
+            $identifier = $update ? $data['uid'] : StringUtility::getUniqueId('NEW');
+
+            $insertData[$importJob->getTable()][$identifier] = $insertArray;
+        }
+
+        /** @deprecated Function call will be removed in 6.0 */
+        $insertData = $this->callHook($insertData, $importJob->getTable());
```

Comment:
See comment in the `callHook()` method. I would suggest to consider dropping the hook directly.